gen/container_test.go

```diff
@@ -1364,6 +1372,29 @@ func TestEmptyContainersRoundTrip(t *testing.T) {
 		assert.Equal(t, give, got)
 	})
 
+	t.Run("required with special treatment of nil slice", func(t *testing.T) {
```

Comment:
can we also add an explicit test to verify that we continue to differentiate between nil and empty for optional list fields?